app/controllers/api/v1/patients_controller.rb

```diff
@@ -1,11 +1,10 @@
 class Api::V1::PatientsController < APIController
-
   def merge_patient(single_patient_params)
-    patient_payload = Api::V1::PatientPayload.new(single_patient_params)
+    patient_payload = Api::V1::PatientPayload.new(single_patient_params.to_hash.with_indifferent_access)
     logger.debug "Patient had errors: #{patient_payload.errors_hash}" if patient_payload.invalid?
     return patient_payload.errors_hash if patient_payload.invalid?
 
-    MergePatientService.new(single_patient_params).merge
+    MergePatientService.new(patient_payload.model_attributes).merge
     nil
```

Comment:
Returning nil is always a code smell. I'd flip the logic here, something like this:

```ruby
  unless patient_payload.invalid?
    MergePatientService.new(patient_payload.model_attributes).merge
  end

  patient_payload.errors_hash
end